### DISCUSSION TITLE
Long time login

### DISCUSSION THREAD

**Discussion by dvorakdominik** [Andre0512/pyhOn#28] [Q&A]
Hi, I'm new to python and I need some advice on how to work with the library.
I would need advice on how to make it so I don't have to constantly log in again because it takes a long time.

Is it possible to save the login in some session and work with the session again and again? Please advise me how?

Thank you

**Comment by kubasaw** [comment]
Haier APIs return `refresh_token` which may be used to speedup login process slightly. Generally, `access_token` and `id_token` issued are valid for 8 hours, but Cognito token and token used to MQTT connection are valid 4 hours.
